server_monitoring/models/server_monitor_process.py

```diff
@@ -0,0 +1,248 @@
+        return result
+
+    openerp.addons.web.controllers.main.DataSet._call_kw = _call_kw
+
+
+class ServerMonitorProcess(models.Model):
+    def __init__(self, pool, cr):
+        super(ServerMonitorProcess, self).__init__(pool, cr)
+
+    _name = 'server.monitor.process'
+
+    def _register_hook(self, cr):
+        global _monkey_patched
+        if _monkey_patched:
+            return
+        _monkey_patched = True
+        _monkey_patch_controller_call_kw()
+        _monkey_patch_object_proxy_execute()
+
+    @api.model
+    def _default_pid(self):
+        return os.getpid()
+
+    @api.model
+    def _default_cpu_time(self):
+        r = resource.getrusage(resource.RUSAGE_SELF)
+        cpu_time = r.ru_utime + r.ru_stime
+        return cpu_time
+
+    @api.model
+    def _default_memory(self):
+        try:
+            rss, vms = psutil.Process(os.getpid()).get_memory_info()
+        except AttributeError:
+            # happens on travis
+            vms = 0
+        return vms
+
+    @api.model
+    def _default_uid(self):
+        return self.env.uid
+
+    @api.model
+    def _default_thread(self):
+        return threading.current_thread().name
+
+    @api.model
+    def _class_count(self):
+        counts = {}
+        context = self.env.context
+        if context.get('_x_no_class_count'):
+            return []
+        if context.get('_x_no_gc_collect'):
+            gc.collect()
+            gc.collect()
+        for obj in gc.get_objects():
+            if isinstance(obj, BLACKLIST):
+                continue
+            try:
+                cls = obj.__class__
+            except:
+                if isinstance(obj, types.ClassType):
+                    cls = types.ClassType
+                else:
+                    _logger.warning('Unknown object type for %r (%s)',
+                                    obj, type(obj))
+                    continue
+            name = '%s.%s' % (cls.__module__, cls.__name__)
+            try:
+                counts[name] += 1
+            except KeyError:
+                counts[name] = 1
+        info = []
+        for name, count in sorted(counts.items(),
+                                  key=itemgetter(1),
+                                  reverse=True):
+            if count < 2:
+                break
+            info.append({'name': name,  'count': count})
+        return [(0, 0, val) for val in info]
+
+    name = fields.Datetime('Timestamp',
+                           default=fields.Datetime.now,
+                           readonly=True)
+    pid = fields.Integer('Process ID',
+                         default=_default_pid,
+                         readonly=True,
+                         group_operator='count')
+    thread = fields.Char('Thread ID',
+                         default=_default_thread,
+                         readonly=True)
+    cpu_time = fields.Float(
+        'CPU time',
+        readonly=True,
+        default=_default_cpu_time,
```

Comment:
Normally [you should define a lambda calling that method, to allow inheritance, as guidelines explain](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#fields). The same applies in many other places.